openhtf/exe/test_state.py

```diff
@@ -116,17 +116,13 @@ def GetFinishedRecord(self):
     """
     self.logger.debug('Finishing test execution with state %s.', self._state)
 
-    if 'config' in self.record.metadata:
-      self.logger.warning('config already set in metadata, not saving config')
-    else:
-      self.record.metadata['config'] = conf.Config()
-
     if not self.record.dut_id:
       raise BlankDutIdError(
           'Blank or missing DUT ID, HTF requires a non-blank ID.')
 
-    return self.record._replace(
-        end_time_millis=util.TimeMillis(), outcome=self._state)
+    self.record.end_time_millis = util.TimeMillis()
```

Comment:
I was a little concerned about this convention in the first place - GetFinishedRecord() was only ever intended to be called once at the end of the lifetime of this instance, though it's not enforced.   At the time, self.record was intentionally not updated to reflect the end_time and outcome to avoid any weird behavior where someone called GetFinishedRecord() and then went back and continued doing something that used self.record (which could result, for example, in things like having log lines past the end_time_millis of the test_record).

Not sure what the best answer here is, really, but it's worth a conversation, I think.  My thoughts are that we should set self.record to None when GetFinishedRecord() is called (and maybe rename that to something like FinalizeRecord()), and then do some checks for self.record being None and raise a specific exception.  That way we can catch as soon as possible if something wonky happens in the usage of a TestState instance.  Thoughts?